internal/controllers/topology/cluster/patches/engine.go

```diff
@@ -139,23 +148,82 @@ func (e *engine) Apply(ctx context.Context, blueprint *scope.ClusterBlueprint, d
 	return nil
 }
 
+// addVariablesForPatch adds variables for a given ClusterClassPatch to the items in the PatchRequest.
+func addVariablesForPatch(blueprint *scope.ClusterBlueprint, desired *scope.ClusterState, req *runtimehooksv1.GeneratePatchesRequest, definitionFrom string) error {
```

Comment:
Q: Do we have test coverage for this code?

As far as I can tell the existing test doesn't care at all about variables.

Maybe it's as easy as adding one more test case which has inline patches which use:
* 1x global variable
* 1x CP variable (used once for ControlPlane once for the ControlPlane InfraMachineTemplate)
* 1x MD variable (used once for BootstrapConfigTemplate and once for InfraMachineTemplate)
* 1x builtin variable

I think this would give us pretty much 100% coverage

Reply:
I added an engine_test which has most of this, but I think some of it is redundant i.e. infra + bootstrap isn't needed, control plane only uses builtin variables etc.

Reply:
I agree with control plane. We had in general no coverage on this level that builtins are working correctly

Apart from that I was hoping that we have test coverage that could proof that we are setting the per-patch user-defined variables for all templates correctly.

Currently we only do this for InfraCluster (for global) and MD.InfraMachineTemplate which leaves us with no coverage for MD.BootstrapConfigTemplate if I"m not mistaken

But I can also do a follow-up PR to improve the coverage if you prefer to not add it in this PR

Reply:
Hm I see what you're saying. We don't distinguish between bootstrapconfig and inframachine in the prod code.

Anyway, thx for the extended coverage!